pkg/estimator/client/general.go

```diff
@@ -33,13 +31,17 @@ func (ge *GeneralEstimator) MaxAvailableReplicas(clusters []*clusterv1alpha1.Clu
 }
 
 func (ge *GeneralEstimator) maxAvailableReplicas(cluster *clusterv1alpha1.Cluster, replicaRequirements *workv1alpha2.ReplicaRequirements) int32 {
-	var maximumReplicas int64 = math.MaxInt32
 	resourceSummary := cluster.Status.ResourceSummary
-
 	if resourceSummary == nil {
 		return 0
 	}
 
+	allowedPodNumber := resourceSummary.Allocatable.Pods().Value() - resourceSummary.Allocated.Pods().Value() - resourceSummary.Allocating.Pods().Value()
+	if allowedPodNumber <= 0 {
+		return 0
+	}
```

Comment:
How is it possible that the pod < 0? I can imagine `allowedPodNumber==0`.
Shall we add a comment on it?

Reply:
When too may pods have been created, they cannot been scheduled due to lack of `pod` resource. So the variable `allocating` may be large.